zerver/tests/test_narrow.py

```diff
@@ -61,206 +68,253 @@ def mute_stream(realm, user_profile, stream_name):
+        # type: () -> None
         term = dict(operator='has', operand='image', negated=True)
         self._do_add_term_test(term, 'WHERE NOT has_image')
 
     def test_add_term_using_has_operator_and_link_operand(self):
+        # type: () -> None
         term = dict(operator='has', operand='link')
         self._do_add_term_test(term, 'WHERE has_link')
 
     def test_add_term_using_has_operator_link_operand_and_negated(self):  # NEGATED
+        # type: () -> None
         term = dict(operator='has', operand='link', negated=True)
         self._do_add_term_test(term, 'WHERE NOT has_link')
 
     def test_add_term_using_has_operator_non_supported_operand_should_raise_error(self):
+        # type: () -> None
         term = dict(operator='has', operand='non_supported')
         self.assertRaises(BadNarrowOperator, self._build_query, term)
 
     def test_add_term_using_in_operator(self):
+        # type: () -> None
         mute_stream(self.realm, self.user_profile, 'Verona')
         term = dict(operator='in', operand='home')
         self._do_add_term_test(term, 'WHERE recipient_id NOT IN (:recipient_id_1)')
 
     def test_add_term_using_in_operator_and_negated(self):
+        # type: () -> None
         # negated = True should not change anything
         mute_stream(self.realm, self.user_profile, 'Verona')
         term = dict(operator='in', operand='home', negated=True)
         self._do_add_term_test(term, 'WHERE recipient_id NOT IN (:recipient_id_1)')
 
     def test_add_term_using_in_operator_and_all_operand(self):
+        # type: () -> None
         mute_stream(self.realm, self.user_profile, 'Verona')
         term = dict(operator='in', operand='all')
         query = self._build_query(term)
         self.assertEqual(str(query), 'SELECT id \nFROM zerver_message')
 
     def test_add_term_using_in_operator_all_operand_and_negated(self):
+        # type: () -> None
         # negated = True should not change anything
         mute_stream(self.realm, self.user_profile, 'Verona')
         term = dict(operator='in', operand='all', negated=True)
         query = self._build_query(term)
         self.assertEqual(str(query), 'SELECT id \nFROM zerver_message')
 
     def test_add_term_using_in_operator_and_not_defined_operand(self):
+        # type: () -> None
         term = dict(operator='in', operand='not_defined')
         self.assertRaises(BadNarrowOperator, self._build_query, term)
 
     def test_add_term_using_near_operator(self):
+        # type: () -> None
         term = dict(operator='near', operand='operand')
         query = self._build_query(term)
         self.assertEqual(str(query), 'SELECT id \nFROM zerver_message')
 
     def _do_add_term_test(self, term, where_clause):
+        # type: (Dict[str, Any], text_type) -> None
         self.assertTrue(where_clause in str(self._build_query(term)))
 
     def _build_query(self, term):
+        # type: (Dict[str, Any]) -> Any
```

Comment:
Where should that be imported from? `zerver/views/messages` as part of the `class` `BadNarrowOperator`?